src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -295,4 +295,49 @@ int32_t GlobalizationNative_EndsWithNative(const uint16_t* localeName, int32_t l
     }
 }
 
+int32_t GlobalizationNative_GetSortKeyNative(const uint16_t* localeName, int32_t lNameLength, const UChar* lpStr, int32_t cwStrLength,
+                                             uint8_t* sortKey, int32_t cbSortKeyLength, int32_t options)
+{
+    @autoreleasepool {
+        if (cwStrLength == 0)
+        {
+            if (sortKey == NULL)
+                sortKey = malloc(1);
+            sortKey[0] = '\0';
+            return 1;
+        }
```

Comment:
I'm not sure I'm following what this function is supposed to do.

Why do we need to allocate sortKey and return `1` when `cwStrLength` is 0?
Callers of this function passing in `null` for `sortKey` wouldn't necessarily free the memory we are allocating here woud they?

Reply:
When `cwStrLength` is 0 , we want to return for empty string valid answer (terminated empty string), otherwise https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs#L831 will consider that there was a failure.